orion/controllers/announce/announce_test.go

```diff
@@ -0,0 +1,122 @@
+	}
+	
+	announce.AnnounceId = "asdf__asdf"
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid announcement id")
+	}
+	
+	announce.AnnounceId = "valid_id_until__here"
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid announcement id")
+	}
+	
+	announce.AnnounceId = "_"
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid announcement id")
+	}
+}
+
+func TestTitle(t *testing.T) {
+	// Success examples
+	announce := Announce{AnnounceId: "ok", Title: "Valid Title", Message: "not blank"}
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+	
+	announce.Title = "New semester tuition"
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+	
+	announce.Title = "Borrowing & Lending"
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+	
+	announce.Title = "1000 Tips on Studying"
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+	
+	announce.Title = "X"
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+	
+	// Failure examples
+	announce.Title = "&&"
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid title")
+	}
+	
+	announce.Title = "lowercase Start"
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid title")
+	}
+	
+	announce.Title = ""
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid title")
+	}
+	
+	announce.Title = "JH %"
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: invalid title")
+	}
+}
+
+func TestMessage(t *testing.T) {
+	// Success examples
+	announce := Announce{AnnounceId: "ok", Title: "Valid Title", Message: "not blank"}
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+	
+	announce.Message = "                 a   "
+	if err := CheckValidAnnouncement(announce); err != nil {
+		t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+	}
+
+	// Failure examples
+	announce.Message = ""
+	if err := CheckValidAnnouncement(announce); err == nil {
+		t.Error("Check was incorrect, got: nil, expected: empty message")
+	}
+	
+	announce.Message = "                                                    "
```

Comment:
Nice, I think it's good to guard against this. So maybe disallow empty strings and strings with no alphabetic letters? This goes for both messages and author.